spring-cloud-dataflow-docs/src/main/asciidoc/task-developer-guide.adoc

```diff
@@ -57,118 +57,233 @@ If you have debug mode turned on, you can get even more information, in a line s
-. On Line 13, add the `@EnableTask` annotation.
-+
-NOTE: If you do not use an IDE, you also need to un-comment lines 6 through 9.
-. Save your changes to `TasklabApplication.java`.
-. Open `src/main/resources/application.properties`.
-. Add the following properties:
-+
+
+While it may seem small, quite a bit is going on. For more about Spring
+Boot specifics, see the
+http://docs.spring.io/spring-boot/docs/current/reference/html/[Spring Boot reference documentation].
+
+Now let's open the `application.properties` file in `src/main/resources`. We need
+to configure two properties in `application.properties`: We need to set the application
+name (which is translated to the task name), and we need to set the logging for Spring
+Cloud Task to `DEBUG` so that we can see what's going on. The following example shows how
+to do both:
+
+[source]
 ----
 logging.level.org.springframework.cloud.task=DEBUG
-spring.application.name=lab2-task
+spring.application.name=timestamp
 ----
-+
-Setting the logging level for the task to `DEBUG` creates output to let us know that the task works.
-Setting the application name lets us store a meaningful name in the database, so that we can find our task in the `TASK_EXECUTION` table.
-. Save your change to `application.properties`.
 
-Now that we have made our changes, we naturally want to run it and see it work. To do so:
+[[getting-started-at-task]]
+==== The @EnableTask annotation
 
-. From the command line, run `mvn clean package`.
-+
-You should see a typical set of Maven build output for a Spring Boot application (with some additional output, because it is now a Spring Cloud Task).
-. From the command line, run `java -jar target/tasklab-0.0.1-SNAPSHOT.jar`.
-+
-You should see output for a typical (though very basic) Spring Cloud Task, including a line similar to (differing only by its timestamp) the following:
-+
-`2018-03-12 15:13:48.930 DEBUG 5202 --- [           main] o.s.c.t.r.support.SimpleTaskRepository   : Creating: TaskExecution{executionId=0, exitCode=null, taskName='lab2-task', startTime=Mon Mar 12 15:13:48 CDT 2018, endTime=null, exitMessage='null', errorMessage='null', arguments=[]}`
+The first non-boot annotation in our example is the `@EnableTask` annotation. This
+class-level annotation tells Spring Cloud Task to bootstrap it's functionality. By
+default, it imports an additional configuration class (`SimpleTaskConfiguration`). This
+additional configuration registers the `TaskRepository` and the infrastructure for its
+use.
+
+In our demo, the `TaskRepository` uses an embedded H2 database to record the results
+of a task. This H2 embedded database is not a practical solution for a production environment, since
+the H2 DB goes away once the task ends. However, for a quick getting-started
+experience, we will use this in our example as well as echoing to the logs what is being updated
+in that repository. In the <<features-configuration>> section (later in this
```

Comment:
We probably can leave out the "Later in this documentation" if we're providing a link to it.